BIDebugEngine/BIDebugEngine/Debugger.cpp

```diff
@@ -250,72 +250,66 @@ void Debugger::onInstruction(DebuggerInstructionInfo& instructionInfo) {
 
 }
 
-void Debugger::dumpStackToRPT(GameState* gs) {
-    if (!gs->get_vm_context()) return;
-    auto& context = *gs->get_vm_context();
-    std::stringstream str;
-
-    bool dumpFullArray = (gs->get_evaluator() && gs->get_evaluator()->local) ? gs->get_evaluator()->local->get_variable("_debug_dumpFullArrays") != nullptr : false;
-
-    str << "Error at " << "L" << context.sdocpos.sourceline << " (" << context.sdocpos.sourcefile << ")\nCallstack:";
-    intercept::sqf::diag_log(str.str());
-    str.str(std::string());
-    for (auto& it : context.callstack) {
-        if (!it) continue;
-
-
-
-        r_string sourceFile;
-        int line{ 0 };
-
-        auto& type = typeid(*it.get());
-        auto hash = type.hash_code();
-
-        switch (hash) {
-#ifdef X64
-        case 0x796333d0f1231802: {
-#else
-        case 0xed08ac32: { //CallStackItemSimple
-#endif
-            auto stackItem = static_cast<const CallStackItemSimple*>(it.get());
 
+namespace {
```

Comment:
Don't like this. Would much prefer std::visit with some lambda's. But I would prefer simple switch over enum even over that.

Reply:
Reasons this is better than a switch statement: 
- no room for mistakes -- if you didn't implement a case you will get an error
- no need to do the casting boiler plate
- no janky fall through bs
- no default catching new cases you didn't intend because you didn't manually check everywhere

Reasons a switch statement is better?